packages/synapse-react-client/src/components/ChallengeDataDownload/ChallengeDataDownload.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback } from 'react'
+import AddCircleTwoToneIcon from '@mui/icons-material/AddCircleTwoTone'
+import { EntityType } from '@sage-bionetworks/synapse-types'
+import { EntityDetailsListSharedProps } from '../EntityFinder/details/EntityDetailsList'
+import { VersionSelectionType } from '../EntityFinder/VersionSelectionType'
+import { EntityFinderHeader } from '../EntityFinder/EntityFinderHeader'
+import { useEntitySelection } from '../EntityFinder/useEntitySelection'
+import { ChallengeEntityChildrenDetails } from './ChallengeEntityChildrenDetails'
+import { Box, Button } from '@mui/material'
+import { useAddFileBatchToDownloadList } from '../../synapse-queries'
+import { displayToast } from '../ToastMessage'
+import { useSynapseContext } from '../../utils'
+
+type ChallengeDataDownloadProps = {
+  parentContainerId: string
+}
+
+export function ChallengeDataDownload({
+  parentContainerId,
+}: ChallengeDataDownloadProps) {
+  const { downloadCartPageUrl } = useSynapseContext()
+  const selectMultiple = true
+  const [selectedEntities, toggleSelection] = useEntitySelection(selectMultiple)
+
+  const { mutate: addBatchToDownloadList } = useAddFileBatchToDownloadList({
+    onSuccess: () => {
+      displayToast(
+        'File(s) were successfully added to your Download Cart.',
+        'success',
+        {
+          primaryButtonConfig: {
+            text: 'View Download Cart',
+            onClick: () => (window.location.href = downloadCartPageUrl),
+          },
+        },
+      )
+    },
+    onError: e => {
+      displayToast(
+        `Unable to add the file to your Download Cart. ${e.reason}`,
+        'danger',
+      )
+    },
+  })
+
+  const isIdSelected = useCallback(
+    (entity: EntityFinderHeader) => {
+      return selectedEntities.has(entity.id)
+    },
+    [selectedEntities],
+  )
+
+  const onAddClick = useCallback(() => {
+    const entities = selectedEntities.toArray().map(entity => {
+      return {
+        fileEntityId: entity[0],
+        versionNumber: entity[1],
+      }
+    })
+    addBatchToDownloadList(entities)
+  }, [selectedEntities, addBatchToDownloadList])
+
+  const sharedProps: EntityDetailsListSharedProps = {
+    versionSelection: VersionSelectionType.REQUIRED,
+    selectColumnType: 'checkbox',
+    enableSelectAll: true,
+    visibleTypes: [EntityType.FILE],
+    selected: selectedEntities,
+    selectableTypes: [EntityType.FILE],
```

Comment:
@jay-hodgson is it true that the challenge data is always exclusively a flat list of files in a folder? never sub-folders or Synapse tables?

Reply:
Based on the [Challenge Information Schema](https://docs.google.com/spreadsheets/d/1iQ8POJASfblzu1VVQX0Ha_SKdC5t1u6Z7P7OJoIsnPM/edit#gid=1130980120) that @lbradic worked out with stakeholders, and his resulting design, each challenge "Task" will have a folder of flat files.  @vpchung , does this align with your expectation (sub-folders and Synapse Tables are unsupported when providing the Training Data for a particular Task)?

Reply:
Yes, that is my expectation.  For any challenge data that will require sub-folders, we will create a zip or tarball.

Thanks for double-checking!